google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.sdk;
+
+import com.google.api.client.repackaged.com.google.common.annotations.VisibleForTesting;
+import com.google.cloud.tools.intellij.appengine.util.CloudSdkUtil;
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.google.cloud.tools.intellij.util.SystemEnvironmentProvider;
+
+import com.intellij.openapi.fileChooser.FileChooserDescriptorFactory;
+import com.intellij.openapi.options.ConfigurationException;
+import com.intellij.openapi.ui.TextFieldWithBrowseButton;
+import com.intellij.openapi.util.Comparing;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.ui.DocumentAdapter;
+import com.intellij.ui.JBColor;
+
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.event.DocumentEvent;
+
+/**
+ * Reusable panel for configuring the path to the Cloud SDK from various contexts.
+ */
+public class CloudSdkPanel {
+
+  private TextFieldWithBrowseButton cloudSdkDirectoryField;
+  private JLabel warningMessage;
+  private JPanel cloudSdkPanel;
+
+  public CloudSdkPanel(@NotNull CloudSdkService cloudSdkService) {
+    warningMessage.setVisible(false);
+
+    final String cloudSdkDirectoryPath
```

Comment:
aha. I did not realize that reset was called on initialization.

On Tue, Aug 2, 2016 at 3:56 PM, Etan Shaul <notifications@github.com> wrote:

> In
> google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/802#discussion_r73225505>
> :
>
> > +import javax.swing.JPanel;
> > +import javax.swing.event.DocumentEvent;
> > +
> > +/**
> > + * Reusable panel for configuring the path to the Cloud SDK from various contexts.
> > + */
> > +public class CloudSdkPanel {
> > +
> > +  private TextFieldWithBrowseButton cloudSdkDirectoryField;
> > +  private JLabel warningMessage;
> > +  private JPanel cloudSdkPanel;
> > +
> > +  public CloudSdkPanel(@NotNull CloudSdkService cloudSdkService) {
> > +    warningMessage.setVisible(false);
> > +
> > +    final String cloudSdkDirectoryPath
>
> Oh. Yea for the cases where the SdkService is configured, it is
> redundantly checking the path. I'll refactor this.
>
> Regarding the second point - the initialization happens via the reset()
> method of that same class which gets called when the panel is loaded after
> construction. I think I can move the path logic to here too.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/802/files/823228e7856c20565bd54c22937666a4a73f7591#r73225505>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHf5HRpzn3ewacAkrSikwJ1UCeYR8933ks5qb6EHgaJpZM4JZ23N>
> .
>